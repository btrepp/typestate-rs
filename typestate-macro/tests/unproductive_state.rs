use typestate::typestate;

#[typestate]
mod m {
    #[automata]
    struct S {}

    #[state]
    struct A {}
}

fn main() {}