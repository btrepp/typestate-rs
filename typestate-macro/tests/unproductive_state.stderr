error: Non-productive state.
 --> $DIR/unproductive_state.rs:9:12
  |
9 |     struct A {}
  |            ^
